doc/sphinx/architecture.md

```diff
@@ -0,0 +1,252 @@
+    ├── mirshell
+    ├── mirshelldecoration
+    ├── mirscene
+    ├── mirgraphics
+    ├── mirreport
+    ├── mirlttng
+    ├── mirlogging
+    ├── mirnullreport
+    ├── mirconsole
+    ├── mirgl
+    └── mirrenderergl
+├── mirplatform
+├── mirwayland
+├── graphics-gbm-kms
+├── graphics-eglstreamkms
+├── graphics-x11
+├── graphics-wayland
+├── input-evdev
+└── renderer-egl-generic
+```
+
+## miral
+- The "Mir Abstraction Layer" (aka `MirAL` or `miral`) is an API that makes Mir easy for compositor-implementers to work with. It provides the core window management concepts and an interface that is more ABI stable than Mir's internal mirserver API.
+- Public-facing
+- Directories:
+  - `include/miral`
+  - `src/miral`
+  
+## miroil
+- A custom *Mir* library for the [Lomiri](https://lomiri.com/) folks
+- This is unimportant for 99% of people
+- Directories:
+  - `include/miroil`
+  - `src/miroil`
+
+## mircore
+- Fundamental data structures like 2D points, Rectangles, File Descriptor, etc.
+- Public-facing
+- Directories:
+  - `include/core`
+  - `src/core`
+  -
+## mircommon
+- Data structures and utility functions like Mir-event building, logging, keymaps, etc.
+- Differs from `mircore` in that the concepts map strictly onto *Mir*'s concepts instead of being general concepts like a Rectangle
+- Public-facing
+- Directories:
+  - `include/common`
+  - `src/common`
+  - `src/include/common`
+
+## mircookie
+- Provides event timestamps for inputs events that are difficult to spoof.
+- Public-facing
+- Directories:
+  - `include/cookie`
+  - `src/cookie`
+  - `src/include/cookie`
+
+## mirserver
+- Provides the core *Mir* runtime, with the entry-point being the `mir::Server` class.
+- The `Server` class has a `DisplayServer` which provides access to the interfaces that that support the server's runtime.
+  - The `ServerConfiguration` provides a bunch of methods called `the_XYZ()` to access these various instances, where "XYZ" can be replaced with the interface of interest.
+- The classes that the `Server` relies on for its runtime can be found organized in separate folders in this directory.
+- Directories:
+  - `include/server`
+  - `src/server`
+  - `src/include/server`
+  
+## mircompositor
+- Included in `mirserver.so`
+- The `Compositor` is responsible for collecting the set of renderables and sending them off to the outputs to be displayed.
+- Directories:
+  - `src/include/server/mir/compositor`
+  - `src/server/compositor`
+
+## mirinput
+- Included in `mirserver.so`
+- Deals with everything input (e.g. keyboard presses, mouse clicks)
+- The `InputManager` provides access to the specific input platform that is running (e.g. X, Wayland, or libinput).
+- The `InputDispatcher` is responsible for taking events bubbled up from the platform and sending them off to the appropriate listeners
+- Directories
+  - `src/include/server/mir/input`
+  - `src/server/input`
+  
+## mirfrontend
+- TODO: How do I describe this one?
```

Comment:
`src/server/frontend` is vestigial - we should probably merge it with `src/server/frontend_wayland`